### ISSUE TITLE
Static linking on Windows?

### ISSUE THREAD

**Issue by theolivenbaum** [TolikPylypchuk/SharpHook#14]
Would it be possible to static link the library to avoid having to include the VC redistributable?

Otherwise the library fails when loading due to the missing vcruntime140.dll file

```
 ---> System.DllNotFoundException: Unable to load DLL 'uiohook' or one of its dependencies: The specified module could not be found. (0x8007007E)
   at SharpHook.Native.UioHook.SetDispatchProc(DispatchProc dispatchProc, IntPtr userData)
   at SharpHook.Reactive.SimpleReactiveGlobalHook.<>c__DisplayClass58_0.<RunAsync>b__0()

```

**Comment by TolikPylypchuk** [comment]
Hi! Thanks for posting this issue! Unfortunately it doesn't appear to be possible. The answers to [this question on StackOverflow](https://stackoverflow.com/questions/729562/how-to-compile-c-sharp-application-with-c-static-library) say that we _must_ use DLLs.

Maybe, the C runtime could be statically linked to the libuiohook library itself (since its being built during the SharpHook build process, and you can check out the build scripts [here](https://github.com/TolikPylypchuk/SharpHook/blob/master/.github/workflows/build.yml)), but I don't have enough knowledge on building libraries written in C to implement this (I just copied the build scripts from libuiohook).

For now I'm going to close this issue as "won't fix", but feel free to continue the discussion.

**Comment by theolivenbaum** [comment]
It should be as simple as adding another flag to the build script:

https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html

Just need to set `MSVC_RUNTIME_LIBRARY` to `MT`

**Comment by TolikPylypchuk** [comment]
Ok, got it - looks pretty easy. I'll try this in the nearest future.

**Comment by TolikPylypchuk** [comment]
I've changed the libuiohook build process on Windows to link the C runtime statically. I don't have a machine without the Visual C++ Redistributable, so could you please test whether it works on your machine? If it does, then I'll release a new version with this change.

You can get the modified libuiohook binary in [this GitHub action result](https://github.com/TolikPylypchuk/SharpHook/actions/runs/2700324891). Download `libuiohook-build`, unarchive it, and go to `windows/<platform>/bin` to find the new `uiohook.dll`.

The libuiohook DLL files are now 10 times larger as a result (~60 KB -> ~600 KB), but I don't think that's going to be a problem for most people.

**Comment by theolivenbaum** [comment]
Thanks! I'll test tomorrow ( the Windows Sandbox is quite useful for this in case you don't know it)

And yes, the size increase is expected, but still better than having to install the vcruntine distributable manually

**Comment by TolikPylypchuk** [comment]
@theolivenbaum did you have a chance to test the newest build of libuiohook?

Turns out that Windows Sandbox comes with a Pro edition of Windows and I have a Home edition :/

**Comment by TolikPylypchuk** [comment]
I've tested this change in a Windows VM which doesn't have Visual C++ Redistributable and it worked just fine. I've released version 3.1.0 which includes this change, so this issue can be closed.
